### PR TITLE
Support CS42L42 + MAX98360A on JSL

### PR DESCRIPTION
**Pull request by brentlu**
It's originated from this patch series:

https://patchwork.kernel.org/project/alsa-devel/patch/20210606004102.26190-2-brent.lu@intel.com/
https://patchwork.kernel.org/project/alsa-devel/patch/20210606004102.26190-3-brent.lu@intel.com/
https://patchwork.kernel.org/project/alsa-devel/patch/20210606004102.26190-4-brent.lu@intel.com/
https://patchwork.kernel.org/project/alsa-devel/patch/20210606004102.26190-5-brent.lu@intel.com/

Thinking that I will need to modify the topology for CS42L42 eventually, why not just review the patches on sof github instead.

### CODE REVIEW

File: sound/soc/intel/boards/sof_cs42l42.c
Comment (plbossart) [inline sound/soc/intel/boards/sof_cs42l42.c:48]: I am not clear on what this means.

Is this an error case, as I read below
````c
case LINK_NONE:
		default:
			dev_err(dev, "invalid link type %d\n", link_type);
			goto devm_err;
````
Is this used as a 'terminator', e.g. so that you could SPK, DMIC, HDMI and no HP?
Is this a placeholder for future extensions?

Comment (brentlu) [reply sound/soc/intel/boards/sof_cs42l42.c:48]: Like you said, LINK_NONE is used as terminator in the SOF_CS42L42_DAILINK macro. The 'case LINK_NONE' here is for the case that LINK_NONE is not used as terminator like this:

SOF_CS42L42_DAILINK(LINK_HP, LINK_DMIC, LINK_HDMI, LINK_NONE, LINK_SPK),

Comment (plbossart) [reply sound/soc/intel/boards/sof_cs42l42.c:48]: Thanks @brentlu, that's quite clever indeed - and that shows I am not :-)

maybe adding some comments would help, just to clarify the intent and error cases.

Comment (brentlu) [reply sound/soc/intel/boards/sof_cs42l42.c:48]: Will do. It also took me a while to figure out why there is a "case LINK_NONE:" there...

**Review by plbossart — DISMISSED** [review]
LGTM @brentlu, need 2nd approver and we are good to go.

**Comment by plbossart** [comment]
@bardliao can you take a look and approve if you are fine with the PR?

File: sound/soc/intel/boards/sof_cs42l42.c
Comment (bardliao) [inline sound/soc/intel/boards/sof_cs42l42.c]: nitpick: two empty line here.

**Review by bardliao — DISMISSED** [review]
LGTM. only a nitpick for checkpatch.

File: sound/soc/intel/boards/sof_cs42l42.c
Comment (brentlu) [reply sound/soc/intel/boards/sof_cs42l42.c]: removed. thanks for review

**Review by bardliao — APPROVED** [review]

**Review by plbossart — APPROVED** [review]

**Pull request merged**
